staging/src/k8s.io/api/storage/v1beta1/types.go

```diff
@@ -166,7 +166,23 @@ type VolumeAttachmentSource struct {
 	// +optional
 	PersistentVolumeName *string `json:"persistentVolumeName,omitempty" protobuf:"bytes,1,opt,name=persistentVolumeName"`
 
-	// Placeholder for *VolumeSource to accommodate inline volumes in pods.
+	// Represents the source location of a in-line volume in a pod.
+	// +optional
+	InlineVolumeSource *InlineVolumeSource `json:"inlineVolumeSource,omitempty" protobuf:"bytes,2,opt,name=inlineVolumeSource"`
+}
+
+// InlineVolumeSource represents the source location of a in-line volume in a pod.
+type InlineVolumeSource struct {
+	// VolumeSource is copied from the pod. It ensures that attacher has enough
+	// information to detach a volume when the pod is deleted before detaching.
+	// Only CSIVolumeSource can be set.
+	// Required.
+	VolumeSource v1.VolumeSource `json:"volumeSource" protobuf:"bytes,1,opt,name=volumeSource"`
```

Comment:
Why use a generic `VolumeSource` type here when only `CSIVolumeSource` is what we want, can we just `CSIVolumeSource` directly?

Reply:
Historically, this was done with the thought of other drivers could use this.  However, since internal volume drivers are on their way out, your comment make sense.